nessie/src/test/java/org/apache/iceberg/nessie/TestParsedTableIdentifier.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.nessie;
+
+import com.dremio.nessie.client.NessieClient;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestParsedTableIdentifier {
+
+
+  @Test
+  public void noMarkings() {
+    String path = "foo";
+    ParsedTableIdentifier pti = ParsedTableIdentifier.getParsedTableIdentifier(path, new HashMap<>());
+    Assert.assertEquals(path, pti.getTableIdentifier().name());
+    Assert.assertNull(pti.getReference());
+    Assert.assertNull(pti.getTimestamp());
+  }
+
+  @Test
+  public void branchOnly() {
+    String path = "foo@bar";
+    ParsedTableIdentifier pti = ParsedTableIdentifier.getParsedTableIdentifier(path, new HashMap<>());
+    Assert.assertEquals("foo", pti.getTableIdentifier().name());
+    Assert.assertEquals("bar", pti.getReference());
+    Assert.assertNull(pti.getTimestamp());
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void timestampOnly() {
+    String path = "foo#baz";
+    ParsedTableIdentifier.getParsedTableIdentifier(path, new HashMap<>());
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void branchAndTimestamp() {
+    String path = "foo@bar#baz";
+    ParsedTableIdentifier.getParsedTableIdentifier(path, new HashMap<>());
+  }
+
+  @Test(expected = IllegalArgumentException.class)
```

Comment:
We prefer using `AssertHelpers.assertThrows` so that state after the exception was thrown can be validated. For example, testing `catalog.createTable(invalid)` would not only check `ValidationException` but also verify that the table was not created.